pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -1015,7 +1076,12 @@ private SegmentCompletionProtocol.Response processConsumedAfterCommitStart(Strin
         // Common case: A different instance is reporting.
         if (offset.compareTo(_winningOffset) == 0) {
           // Wait until winner has posted the segment before asking this server to KEEP the segment.
-          response = hold(instanceId, offset);
+          // Keep if it's pauseless enabled
```

Comment:
We allow the replicas to build the segment. This ensures that:
1. `Reduced need for disaster recovery`: The server is persisted to disk on atleast one of the servers.
2. `Continued ingestion on replicas`: The replicas can continue ingesting newly created segments rather than waiting for the committing server to complete the commit (existing behavior).